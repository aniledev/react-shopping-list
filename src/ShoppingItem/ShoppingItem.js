import React from "react";

export default function ShoppingItem(props) {
  return (
    <li>
      <h2
        // use a ternary operator; if the props.item.checked evalutes to true then apply the line-through text-declaraction, if it doesn't evaluate to true then apply null
        style={{
          textDecoration: props.item.checked ? "line-through" : null,
        }}
      >
        {props.item.name}
      </h2>
      <button type="button" onClick={props.onCheckItem}>
        check
      </button>
      <button type="button" onClick={props.DeleteItem}>
        delete
      </button>
    </li>
  );
}

ShoppingItem.defaultProps = {
  item: {},
};
