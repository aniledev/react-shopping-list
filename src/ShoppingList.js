import React from "react";

export default class ShoppingList extends React.Component {
  render() {
    return <div></div>;
  }
}
