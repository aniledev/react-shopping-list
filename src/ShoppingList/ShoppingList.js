import React from "react";

// an individual item within the list that contains a check button and delete button. There will be 0 or more of these
export default class ShoppingList extends React.Component {
  render() {
    return <div></div>;
  }
}
