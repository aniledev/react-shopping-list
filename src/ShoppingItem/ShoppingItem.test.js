import React from "react";
import ReactDOM from "react-dom";
import ShoppingItem from "./ShoppingItem";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ShoppingList from "../ShoppingList/ShoppingList";

describe("ShoppingItem Component", () => {
  // start with a smoke test inside a describe block; does the component render properly without errors
  it("should render without errors", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ShoppingItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //SNAPSHOT TESTS BELOW USING ENZYME
  it("should render empty given no item", () => {
    // nothing should render
    const wrapper = shallow(<ShoppingItem />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render the item when supplied", () => {
    // we're adding props to the ShoppingItem for different test cases
    // create a test item with a name and checked key
    const testItem = { name: "test-item", checked: false };
    // this test item should be passed in as a prop to the ShoppingItem component
    const wrapper = shallow(<ShoppingItem item={testItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should strike through checked items", () => {
    // pass in a prop where the checked is true
    const testItem = { name: "checked-item", checked: true };
    const wrapper = shallow(<ShoppingItem item={testItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
