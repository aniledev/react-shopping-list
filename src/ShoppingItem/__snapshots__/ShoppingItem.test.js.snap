// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingItem Component should render empty given no item 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  />
  <button
    type="button"
  >
    check
  </button>
  <button
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component should render the item when supplied 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": null,
      }
    }
  >
    test-item
  </h2>
  <button
    type="button"
  >
    check
  </button>
  <button
    type="button"
  >
    delete
  </button>
</li>
`;

exports[`ShoppingItem Component should strike through checked items 1`] = `
<li>
  <h2
    style={
      Object {
        "textDecoration": "line-through",
      }
    }
  >
    checked-item
  </h2>
  <button
    type="button"
  >
    check
  </button>
  <button
    type="button"
  >
    delete
  </button>
</li>
`;
